### PR TITLE
More warning/error messages/hints to help

### PR DESCRIPTION
**Pull request by adswa**
- Add receive.denyCurrentBranch configuration  as an alternative to enable pushing to non-bare repositories

### CODE REVIEW

File: docs/basics/101-135-help.rst [hunk 1]
```diff
@@ -235,7 +241,22 @@ because remotes are not available:
...
+directory is used instead of ``PWD``, and this is announced with a warning
+similar to this:
+
+.. code-block::
+
+   [WARNING] realpath of PWD=</some/path/to/symlinked/file> is </resolved/path/to/symlinked/file> whenever os.getcwd()=<path/to/dataset>.
+   From now on will be returning os.getcwd(). Directory symlinks in the paths will be resolved.
+
+There is no action to be taken based on this warning. Usually, everything should
+be fine.
```
Comment (mih) [inline docs/basics/101-135-help.rst:259]: I hope we can avoid to have this: https://github.com/datalad/datalad/issues/4425

Comment (adswa) [reply docs/basics/101-135-help.rst:259]: sure. :)

Comment (adswa) [reply docs/basics/101-135-help.rst]: This is removed now. :)

**Comment by adswa** [comment]
I'm assuming this is good to go, will merge. :)

**Pull request merged**